### PR TITLE
10.25.1 changelog

### PR DESCRIPTION
**Pull request by adreed-msft**

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

**Review by siminsavani-msft — APPROVED** [review]

File: ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,17 @@
 
 # Change Log
 
+## Version 10.25.1
+
+### Security fixes
+
+1. Updated Golang to 1.22.4 to address security vulnerabilities
```
Comment (gapra-msft) [inline ChangeLog.md:8]: Should we include this too in the release? https://github.com/Azure/azure-storage-azcopy/pull/2713

Comment (seanmcc-msft) [reply ChangeLog.md:8]: @gapra-msft would the Go SDK have to release first?  Or is that not how it works in Go.

Comment (gapra-msft) [reply ChangeLog.md:8]: its been released already https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azidentity/CHANGELOG.md#160-2024-06-10

Comment (gapra-msft) [reply ChangeLog.md:8]: We dont have to wait for azblob to upgrade I dont think

Comment (gapra-msft) [reply ChangeLog.md:8]: I guess this isn't a blocking step to release, so we can ignore for now and plan to have this in the next release.

Comment (seanmcc-msft) [reply ChangeLog.md:8]: @gapra-msft, we don't have to wait for AzBlob.  We just need CI to finish on https://github.com/Azure/azure-storage-azcopy/pull/2713 and we can include it in the release.

**Pull request merged**